provisioner/windows-restart/provisioner.go

```diff
@@ -263,7 +263,7 @@ var waitForCommunicator = func(ctx context.Context, p *Provisioner) error {
 				cmdKeyCheck.Stdout = &buf
 				cmdKeyCheck.Stdout = io.MultiWriter(cmdKeyCheck.Stdout, &buf2)
 
-				err := p.comm.Start(ctx, cmdKeyCheck)
+				err := cmdKeyCheck.RunWithUi(ctx, p.comm, p.ui)
```

Comment:
RunWithUi already calls Wait() on the command, which renders the wait below in line 271 useless. If we're going to switch to RunWithUi, we should remove that to make sure it doesn't cause any issues.